packages/hardhat-core/test/internal/hardhat-network/helpers/retrieveForkBlockNumber.ts

```diff
@@ -1,9 +1,8 @@
-import { HardhatNetworkProvider } from "../../../../src/internal/hardhat-network/provider/provider";
-
 /* eslint-disable @typescript-eslint/dot-notation */
 
+// EDR-TODO: this should be adapted or removed
 export async function retrieveLatestBlockNumber(
-  provider: HardhatNetworkProvider
+  provider: any
```

Comment:
Can't we just use the `eth_blockNumber` API for this?

Reply:
Maybe? To be honest, I didn't want to spend any time on this since we'll have to do it properly later anyway, and I want to have the provider working asap since it's a big blocker for all of us.